src/Illuminate/Filesystem/functions.php

```diff
@@ -7,10 +7,10 @@
      * Join the given paths together.
      *
      * @param  string|null  $basePath
-     * @param  string  ...$paths
+     * @param  string|null  ...$paths
      * @return string
      */
-    function join_paths($basePath, string ...$paths)
+    function join_paths($basePath, ?string ...$paths)
```

Comment:
```suggestion
    function join_paths($basePath, ...$paths)
```

Removing typed hint is better than accepting null. As it never intended to.